### PR TITLE
Detect if bridge network is same with other networks

### LINKED ISSUE #1430: VCH loops reloading port-layer - consumes 100% CPU
I've seen this a few times on the vSAN cluster. This was just running a simple docker info test.
Here is the looping output from init.log
```
time="2016-07-10T06:25:31Z" level=info msg="Reload triggered" 
time="2016-07-10T06:25:31Z" level=info msg="Triggered reload" 
time="2016-07-10T06:25:31Z" level=info msg="Loading main configuration" 
time="2016-07-10T06:25:31Z" level=info msg="Loaded executor config" 
time="2016-07-10T06:25:31Z" level=warning msg="Unable to add additional alias on link client for bridge" 
time="2016-07-10T06:25:31Z" level=info msg="setting ip address 10.118.67.211/23 for link client" 
time="2016-07-10T06:25:31Z" level=warning msg="address 10.118.67.211/23 already set on interface client" 
time="2016-07-10T06:25:31Z" level=info msg="Added nameservers: [10.118.65.1 10.118.65.2]" 
time="2016-07-10T06:25:31Z" level=info msg="setting ip address 10.118.67.211/23 for link client" 
time="2016-07-10T06:25:31Z" level=warning msg="address 10.118.67.211/23 already set on interface client" 
time="2016-07-10T06:25:31Z" level=info msg="updated default route to external interface, gateway: 10.118.67.253" 
time="2016-07-10T06:25:31Z" level=info msg="Added nameservers: [10.118.65.1 10.118.65.2]" 
time="2016-07-10T06:25:31Z" level=info msg="setting ip address 10.118.67.211/23 for link client" 
time="2016-07-10T06:25:31Z" level=warning msg="address 10.118.67.211/23 already set on interface client" 
time="2016-07-10T06:25:31Z" level=info msg="Added nameservers: [10.118.65.1 10.118.65.2]" 
time="2016-07-10T06:25:31Z" level=error msg="unable to encode []net.IP (slice) for guestinfo./networks|external/network/dns: net.IP is an unhandled type" 
time="2016-07-10T06:25:31Z" level=error msg="unable to encode []net.IP (slice) for guestinfo./networks|management/network/dns: net.IP is an unhandled type" 
time="2016-07-10T06:25:31Z" level=error msg="unable to encode []net.IP (slice) for guestinfo./networks|client/network/dns: net.IP is an unhandled type" 
time="2016-07-10T06:25:31Z" level=warning msg="Re-launching process for session port-layer (count: 182)" 
time="2016-07-10T06:25:31Z" level=info msg="opening /var/log/vic/port-layer.log for session logging" 
time="2016-07-10T06:25:33Z" level=info msg="port-layer exit code: 1" 
time="2016-07-10T06:25:33Z" level=error msg="unable to encode []*metadata.ExitLog (slice) for guestinfo..sessions|port-layer.diagnostics.exitlogs: *metadata.ExitLog is an unhandled type" 
time="2016-07-10T06:25:33Z" level=info msg="Reload triggered" 
```
port-layer.log
```
time="2016-07-10T06:27:06Z" level=info msg="Launching portlayer server server on 127.0.0.1:6063" 
time="2016-07-10T06:27:06Z" level=debug msg="Logging into extension com.vmware.vic.423e4a97-8283-e767-d580-df14b4763433" 
time="2016-07-10T06:27:06Z" level=debug msg="Session Environment Info: " API Type=VirtualCenter API Version=6.0 Build=2776510 Name="VMware vCenter Server" OS Type=linux-x64 Product ID=vpx UUID=5ff06379-3d54-4c43-b363-c1c41f5df902 Vendor="VMware, Inc." Version=6.0.0 
time="2016-07-10T06:27:06Z" level=debug msg="vSphere resource cache populating..." 
time="2016-07-10T06:27:06Z" level=debug msg="Cached dc: ha-datacenter" 
time="2016-07-10T06:27:06Z" level=debug msg="Cached cluster: /ha-datacenter/host/vSAN" 
time="2016-07-10T06:27:06Z" level=debug msg="Cached ds: vsanDatastore" 
time="2016-07-10T06:27:07Z" level=debug msg="Cached pool: /ha-datacenter/host/vSAN/Resources" 
time="2016-07-10T06:27:07Z" level=debug msg="vSphere resource cache populated..." 
time="2016-07-10T06:27:07Z" level=debug msg="Decoded VCH config for execution: exec.Configuration{DebugLevel:0, ContainerVMSize:metadata.Resources{CPU:types.ResourceAllocationInfo{DynamicData:types.DynamicData{}, Reservation:0, ExpandableReservation:(*bool)(nil), Limit:0, Shares:(*types.SharesInfo)(nil), OverheadLimit:0}, Memory:types.ResourceAllocationInfo{DynamicData:types.DynamicData{}, Reservation:0, ExpandableReservation:(*bool)(nil), Limit:0, Shares:(*types.SharesInfo)(nil), OverheadLimit:0}, IO:types.ResourceAllocationInfo{DynamicData:types.DynamicData{}, Reservation:0, ExpandableReservation:(*bool)(nil), Limit:0, Shares:(*types.SharesInfo)(nil), OverheadLimit:0}, Storage:types.ResourceAllocationInfo{DynamicData:types.DynamicData{}, Reservation:0, ExpandableReservation:(*bool)(nil), Limit:0, Shares:(*types.SharesInfo)(nil), OverheadLimit:0}}, ContainerStores:[]url.URL(nil), ComputeResources:[]types.ManagedObjectReference{types.ManagedObjectReference{Type:\"ResourcePool\", Value:\"resgroup-1644\"}}, ResourcePool:(*object.ResourcePool)(0xc8209d7000), BootstrapImagePath:url.URL{Scheme:\"\", Opaque:\"\", User:(*url.Userinfo)(nil), Host:\"\", Path:\"\", RawPath:\"\", RawQuery:\"\", Fragment:\"\"}, ContainerNameConvention:\"\", Networks:map[string]*metadata.NetworkEndpoint{\"bridge\":(*metadata.NetworkEndpoint)(0xc8203ea2c0), \"client\":(*metadata.NetworkEndpoint)(0xc8203ea420), \"external\":(*metadata.NetworkEndpoint)(0xc8203ea6e0), \"management\":(*metadata.NetworkEndpoint)(0xc8203eab00)}, DebugNetwork:object.NetworkReference(nil)}" 
time="2016-07-10T06:27:07Z" level=debug msg="Decoded VCH config for network: network.Configuration{BridgeNetwork:\"bridge\", ContainerNetworks:map[string]*network.ContainerNetwork{\"bridge\":(*network.ContainerNetwork)(0xc8206b1c80)}, BridgeLink:network.Link(nil)}" 
time="2016-07-10T06:27:07Z" level=debug msg="Decoded VCH config for storage: storage.Configuration{DebugLevel:0, ImageStores:[]url.URL{url.URL{Scheme:\"ds://\", Opaque:\"\", User:(*url.Userinfo)(0xc8205e8e40), Host:\"vsanDatastore\", Path:\"vsanDatastore\", RawPath:\"\", RawQuery:\"\", Fragment:\"\"}}, VolumeLocations:map[string]url.URL{}}" 
time="2016-07-10T06:27:07Z" level=debug msg="Logging into extension com.vmware.vic.423e4a97-8283-e767-d580-df14b4763433" 
time="2016-07-10T06:27:07Z" level=debug msg="Session Environment Info: " API Type=VirtualCenter API Version=6.0 Build=2776510 Name="VMware vCenter Server" OS Type=linux-x64 Product ID=vpx UUID=5ff06379-3d54-4c43-b363-c1c41f5df902 Vendor="VMware, Inc." Version=6.0.0 
time="2016-07-10T06:27:07Z" level=debug msg="vSphere resource cache populating..." 
time="2016-07-10T06:27:07Z" level=debug msg="Cached dc: ha-datacenter" 
time="2016-07-10T06:27:07Z" level=debug msg="Cached cluster: /ha-datacenter/host/vSAN" 
time="2016-07-10T06:27:07Z" level=debug msg="Cached ds: vsanDatastore" 
time="2016-07-10T06:27:08Z" level=debug msg="Cached pool: /ha-datacenter/host/vSAN/Resources" 
time="2016-07-10T06:27:08Z" level=debug msg="vSphere resource cache populated..." 
time="2016-07-10T06:27:08Z" level=debug msg="Looking for scsi controller with label SCSI0" 
time="2016-07-10T06:27:08Z" level=debug msg="Found pvscsi controller directory: 0000:03:00.0" 
time="2016-07-10T06:27:08Z" level=debug msg="Disk location format: /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:%d:0" 
time="2016-07-10T06:27:08Z" level=info msg="Created store parent directory (VIC) at [vsanDatastore] VIC" 
time="2016-07-10T06:27:08Z" level=info msg="Datastore path is [vsanDatastore] VIC" 
time="2016-07-10T06:27:08Z" level=debug msg="Looking for scsi controller with label SCSI0" 
time="2016-07-10T06:27:08Z" level=debug msg="Found pvscsi controller directory: 0000:03:00.0" 
time="2016-07-10T06:27:08Z" level=debug msg="Disk location format: /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:%d:0" 
time="2016-07-10T06:27:08Z" level=info msg="Refreshing volume cache." 
time="2016-07-10T06:27:08Z" level=fatal msg="failed to create network context: Link alias bridge not found" 
time="2016-07-10T06:27:09Z" level=info msg="Launching portlayer server server on 127.0.0.1:6063" 
```

### PR DESCRIPTION
**Pull request by emlin**
Fixes #1430

### CODE REVIEW

File: lib/install/validate/validator.go [hunk 1]
```diff
@@ -277,6 +277,10 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *metadat
 			Default: true, // external network is default for appliance
 		},
 	})
+	// Bridge network should be different with all other networks
+	if input.BridgeNetworkName == input.ExternalNetworkName {
+		v.NoteIssue(errors.Errorf("bridge network %s is used for external network, please choose different network", input.BridgeNetworkName))
```
Comment (hickeng) [inline lib/install/validate/validator.go]: the bridge network must not be shared with another network role - external also uses %s

File: lib/install/validate/validator.go [hunk 2]
```diff
@@ -295,6 +299,9 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *metadat
 			},
 		},
 	})
+	if input.BridgeNetworkName == input.ClientNetworkName {
+		v.NoteIssue(errors.Errorf("bridge network %s is used for client network, please choose different network", input.BridgeNetworkName))
```
Comment (hickeng) [inline lib/install/validate/validator.go]: the bridge network must not be shared with another network role - client also uses %s

File: lib/install/validate/validator.go [hunk 3]
```diff
@@ -310,6 +317,9 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *metadat
 			},
 		},
 	})
+	if input.BridgeNetworkName == input.ManagementNetworkName {
+		v.NoteIssue(errors.Errorf("bridge network %s is used for management network, please choose different network", input.BridgeNetworkName))
```
Comment (hickeng) [inline lib/install/validate/validator.go]: the bridge network must not be shared with another network role - management also uses %s

File: lib/install/validate/validator.go [hunk 4]
```diff
@@ -404,6 +414,9 @@ func (v *Validator) network(ctx context.Context, input *data.Data, conf *metadat
 			Nameservers: dns,
 			Pools:       pools,
 		}
+		if input.BridgeNetworkName == net {
+			v.NoteIssue(errors.Errorf("bridge network %s is used for container network %s, please choose different network", input.BridgeNetworkName, name))
```
Comment (hickeng) [inline lib/install/validate/validator.go]: the bridge network must not be shared with another network role - %s also mapped as container network %s

**Comment by hickeng** [comment]
LGTM

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by emlin** [comment]
♻️

**Pull request merged**